db/background_mgr.go

```diff
@@ -290,43 +586,103 @@ func (t *TombstoneCompactionManager) ResetStatus() {
 type AttachmentCompactionManager struct {
 	MarkedAttachments base.AtomicInt
 	PurgedAttachments base.AtomicInt
+	CompactID         string
+	Phase             string
 	lock              sync.Mutex
 }
 
 var _ BackgroundManagerProcessI = &AttachmentCompactionManager{}
 
-func NewAttachmentCompactionManager() *BackgroundManager {
+func NewAttachmentCompactionManager(bucket base.Bucket) *BackgroundManager {
 	return &BackgroundManager{
-		Process:    &AttachmentCompactionManager{},
+		Process: &AttachmentCompactionManager{},
+		clusterAwareOptions: &ClusterAwareBackgroundManagerOptions{
+			bucket:        bucket,
+			processSuffix: "compact",
+		},
 		terminator: make(chan struct{}, 1),
 	}
 }
 
-func (a *AttachmentCompactionManager) Run(options map[string]interface{}, terminator chan struct{}) error {
-	database := options["database"].(*Database)
+func (a *AttachmentCompactionManager) Init(clusterStatus []byte) error {
+	if clusterStatus != nil {
+		var attachmentResponseStatus AttachmentManagerResponse
+		err := base.JSONUnmarshal(clusterStatus, &attachmentResponseStatus)
+
+		// If the previous run completed, or there was an error during unmarshalling the status we will start the
+		// process from scratch with a new compaction ID. Otherwise, we should resume with the compact ID and phase
```

Comment:
Do we have a way to force a restart from scratch if either the compact was manually stopped, or the node failed?

Reply:
Added this